mirrord/cli/src/main.rs

```diff
@@ -415,6 +415,12 @@ where
         .map(|(k, v)| CString::new(format!("{k}={v}")))
         .collect::<CliResult<Vec<_>, _>>()?;
 
+    // upon success, print all messages in the progress buffer
+    let buffer = progress.success_and_return_buffer(Some("Ready!"));
```

Comment:
Only the `SpinnerProgress` actually implements this right?

So why not do something like, in the `success` impl of the `SpinnerProgress`, you add this after the `finish_with_message`? You could remove this function (`success_and_return_buffer`), and just handle this stuff in `success`. wdyt?